src/rounded_rect.rs

```diff
@@ -0,0 +1,138 @@
+//! A rectangle with rounded corners.
+
+use crate::{PathEl, Rect, Shape, Vec2};
+use std::f64::consts::PI;
+
+/// A rectangle with rounded corners.
+#[derive(Clone, Copy, Default, Debug)]
+pub struct RoundedRect {
+    /// Coordinates of the rectangle.
+    pub rect: Rect,
+    /// Radius of all four corners.
+    pub radius: f64,
+}
+
+impl RoundedRect {
+    /// A new rectangle from minimum and maximum coordinates.
+    #[inline]
+    pub fn new(x0: f64, y0: f64, x1: f64, y1: f64, radius: f64) -> RoundedRect {
+        RoundedRect {
+            rect: Rect::new(x0, y0, x1, y1),
+            radius,
+        }
+    }
+
+    /// A new rectangle from two points.
+    ///
+    /// The result will have non-negative width and height.
+    #[inline]
+    pub fn from_points(p0: Vec2, p1: Vec2, radius: f64) -> RoundedRect {
+        RoundedRect {
+            rect: Rect::from_points(p0, p1),
+            radius,
+        }
+    }
+
+    /// A new rectangle from origin and size.
+    ///
+    /// The result will have non-negative width and height.
+    #[inline]
+    pub fn from_origin_size(origin: Vec2, size: Vec2, radius: f64) -> RoundedRect {
+        RoundedRect {
+            rect: Rect::from_points(origin, origin + size),
+            radius,
+        }
+    }
+
+    /// The width of the rectangle.
+    ///
+    /// Note: nothing forbids negative width.
+    #[inline]
+    pub fn width(&self) -> f64 {
+        self.rect.width()
+    }
+
+    /// The height of the rectangle.
+    ///
+    /// Note: nothing forbids negative height.
+    #[inline]
+    pub fn height(&self) -> f64 {
+        self.rect.height()
+    }
+
+    /// The origin of the vector.
+    ///
+    /// This is the top left corner in a y-down space and with
+    /// non-negative width and height.
+    #[inline]
+    pub fn origin(&self) -> Vec2 {
+        self.rect.origin()
+    }
+
+    /// The center point of the rectangle.
+    #[inline]
+    pub fn center(&self) -> Vec2 {
+        self.rect.center()
+    }
+
+    /// Take absolute value of width, height and radius.
+    ///
+    /// The resulting rect has the same extents as the original, but is
```

Comment:
What happens when the radius has different sign as the area? I'm wondering whether this case should be forbidden by construction.